.github/workflows/contracts.yml

```diff
@@ -0,0 +1,67 @@
+name: Cosmwasm Contracts
+on:
+  pull_request:
+  push:
+
+
+jobs:
+  test:
+    name: Test Suite
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        workdir: [./x/ibc-rate-limit]
+#        output: [./x/ibc-rate-limit/testdata/rate_limit.wasm]
+
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v2
+
+      - name: Install toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: nightly
+          target: wasm32-unknown-unknown
+
+      - name: Build
+        working-directory: ${{ matrix.workdir }}
+        run: >
+          rustup target add wasm32-unknown-unknown;
+          cargo build --release --target wasm32-unknown-unknown
+
+      - name: Test
+        working-directory: ${{ matrix.workdir }}
+        run: >
+          cargo test
+
+#      - name: Check Test Data
```

Comment:
Is this going to be re-enabled in a future PR?

Reply:
yes, this doesn't make sense in the current PR since the testdata isn't there (that's part of the go tests)

One issue I still need to figure out is what the best way to compile the contract in the GH action is. Right now we're testing the build in dev mode, which is faster and has no optimizations. To build the wasm code that can be uploaded to the chain (the one in `testdata/rate_limiter.wasm`) we need to optimize by either using https://github.com/CosmWasm/rust-optimizer/ or https://github.com/mandrean/cw-optimizoor. The first one adds complexity to the task cause we need to setup docker inside the image, and the second one takes a long time to install (and I didn't want to add a workflow that takes 10+ minutes).

Right now, I'm leaning towards using cw-optimizoor and caching the install step, but I'm not very familiar with how caching works on GH workfows, so I have to look into that.

Reply:
Added the optimizations and data testing is uncommented in the main branch (will be uncommented in a future PR). Marking this as resolved for now.